File: search/leetCode_firstBadVersion.java
package binarySearch;

/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */
/*
    public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int start = 0;
        int end = n ;
        
        while (start <= end ){
            int mid = start + (end - start)/2;
            if ( isBadVersion(mid)) end = mid-1;
            else start = mid+1;
        }
        return start; 
    }
}
*/

File: search/leetCode_firstBadVersion.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):
'''
class Solution(object):
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        l=1
        r=n
        while l < r:
            mid= l + (r-l)//2
            if isBadVersion(mid):
                r= mid
            else: 
                l=mid+1
                
        return l
'''